parsing/mod_adapt.py

```diff
@@ -0,0 +1,104 @@
+"""
+This module contains functionality for extracting a grammar from
+classes in a module.
+"""
+import types
+from parsing.grammar import Precedence, TokenSpec, NontermSpec, SpecError
+from parsing.ast_objs import Token, Nonterm
+
+class ModuleAdapter(object):
+    def __init__(self, modules):
+        if type(modules) == types.ModuleType:
+            # Wrap single module in a list.
+            modules = [modules]
+        self.modules = modules
+        items = []
+        for module in self.modules:
+            for k, v in module.__dict__.items():
+                if isinstance(v, type) and isinstance(v.__doc__, str):
+                    dirtoks = v.__doc__.split(" ")
+                    items.append((module, k, v, dirtoks))
+        self.named_objs = items
+
+    def get_precedences(self):
+        result = []
```

Comment:
Thanks for having a look!
Arguably these are much less expensive than the subsequent LR automaton calculations and/or pickling of the parsing tables. This is still a good idea because TokenSpec/NontermSpec get unique IDs and caching means that different consumers (parser spec + scanner) get TokenSpec/NontermSpec instances with matching IDs.